modules/apps/29-fee/keeper/msg_server.go

```diff
@@ -17,6 +17,12 @@ var _ types.MsgServer = Keeper{}
 func (k Keeper) RegisterCounterpartyAddress(goCtx context.Context, msg *types.MsgRegisterCounterpartyAddress) (*types.MsgRegisterCounterpartyAddressResponse, error) {
 	ctx := sdk.UnwrapSDKContext(goCtx)
 
+	// only register counterparty address if the channel exists
+	_, found := k.channelKeeper.GetChannel(ctx, msg.PortId, msg.ChannelId)
```

Comment:
can we add a check to see if the channel is fee enabled as well?

Reply:
Yup I can do that. I was going to mention this also. I feel like we discussed this and said not to add it for some reason but I can't remember why (can't think of any reason why not also)

Reply:
added